### PR TITLE
fix: add expected fail for encryption key tests on iOS 15 simulator and bump components SQPIT-798

### PR DESCRIPTION
**Pull request by zenkins**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQPIT-798" title="SQPIT-798" target="_blank">TaskSQPIT-798</a>  Azure: data model/ sync engine test crashed but report as passed
  </td></table>
  <br />
 

Update tests for https://github.com/wireapp/wire-ios-data-model/pull/1308. 
We can verify the expected fails are only occurs on iOS 15 by comparing the test result on iOS 14 simulator.

### CODE REVIEW

**Review by billypchan — APPROVED** [review]

**Comment by billypchan** [comment]
failing tests:

```
	UnitTests:
		EventDecoderTest.testThatItProcessesEventsWithEncryptionKeys()
		StoreUpdateEventTests.testThatItCanConvertAnEventToStoredEventAndBack()
		StoreUpdateEventTests.testThatItCanNotConvertEncryptedStoredEventWithoutPrivateKey()
		StoreUpdateEventTests.testThatItConvertsStoreEventToTheUpdateEventIfThePrivateKeyIsNil()
		StoreUpdateEventTests.testThatItDecryptsAndConvertsStoreEventToTheUpdateEventIfThePrivateKeyIsNotNil()
		StoreUpdateEventTests.testThatItDoesNotEncryptEventIfThePublicKeyIsNil()
		StoreUpdateEventTests.testThatItEncryptsEventIfThePublicKeyIsNotNil()
		StoreUpdateEventTests.testThatItFetchesAllStoredEvents()
		StoreUpdateEventTests.testThatItOrdersEventsBySortIndex()
		StoreUpdateEventTests.testThatItReturnsHighestIndex()
		StoreUpdateEventTests.testThatItReturnsOnlyDefinedBatchSize()
		StoreUpdateEventTests.testThatYouCanCreateAnEvent()
		ZMLocalNotificationTests_Message.testThatItGeneratesCorrectCategoryIfEncryptionAtRestIsEnabledForNormalUser()
		ZMLocalNotificationTests_Message.testThatItGeneratesCorrectCategoryIfEncryptionAtRestIsEnabledForTeamUser()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsLocked_AfterBackgroundTaskCompletesInTheBackground()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsLocked_AfterEnteringBackground()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsLocked_WhenTheCustomTimeoutHasExpiredInTheBackground()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsNotLocked_IfThereIsAnActiveBackgroundTask()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsUnlocked_AfterActivatingEncryptionAtRest()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsUnlocked_AfterDeactivatingEncryptionAtRest()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsUnlocked_AfterUnlockingDatabase()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseIsUnlocked_WhenEncryptionAtRestIsDisabled()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseLockedHandlerIsCalled_AfterDatabaseIsLocked()
		ZMUserSessionTests_EncryptionAtRest.testThatDatabaseLockedHandlerIsCalled_AfterUnlockingDatabase()
		ZMUserSessionTests_EncryptionAtRest.testThatDelegateIsCalled_WhenEncryptionAtRestIsDisabled()
		ZMUserSessionTests_EncryptionAtRest.testThatDelegateIsCalled_WhenEncryptionAtRestIsEnabled()
		ZMUserSessionTests_EncryptionAtRest.testThatIfDatabaseIsLocked_ThenUserSessionLockIsSet()
		ZMUserSessionTests_EncryptionAtRest.testThatIfDatabaseIsNotLocked_ThenUserSessionLockIsNotSet()
		ZMUserSessionTests_EncryptionAtRest.testThatOldEncryptionKeysAreReplaced_AfterActivatingEncryptionAtRest()
	IntegrationTests:
		SessionManagerTests_EncryptionAtRestIsEnabledByDefault_Option.testThatEncryptionAtRestIsEnabled_OnActiveUserSession()
		SessionManagerTests_EncryptionAtRestMigration.testThatDatabaseIsMigrated_WhenEncryptionAtRestIsDisabled()
		SessionManagerTests_EncryptionAtRestMigration.testThatDatabaseIsMigrated_WhenEncryptionAtRestIsEnabled()
```

**Comment by billypchan** [comment]
failing tests with testing with iOS 14.4 simulator
```
Failing tests:
	UnitTests:
		UserImageAssetUpdateStrategyTests.testThatItCreatesRequestWithExpectedData()
```

**Comment by billypchan** [comment]
test passed with iOS 14 simulator. I will apply fix for iOS 15 simulator

**Review by sb88k — APPROVED** [review]

**Comment by billypchan** [comment]
failed test `testThatDatabaseIsMigrated_WhenEncryptionAtRestIsDisabled `

**Comment by billypchan** [comment]
still failing tests
```
Failing tests:
	IntegrationTests:
		SessionManagerTests_EncryptionAtRestMigration.testThatDatabaseIsMigrated_WhenEncryptionAtRestIsDisabled()
		SessionManagerTests_EncryptionAtRestMigration.testThatDatabaseIsMigrated_WhenEncryptionAtRestIsEnabled()
```

**Comment by billypchan** [comment]
failing tests
```
Failing tests:
	IntegrationTests:
		SessionManagerTests_EncryptionAtRestIsEnabledByDefault_Option.testThatEncryptionAtRestIsEnabled_OnActiveUserSession()
		SessionManagerTests_EncryptionAtRestMigration.testThatDatabaseIsMigrated_WhenEncryptionAtRestIsDisabled()
		UserTests_AccountDeletion.testThatUserIsMarkedAsDeleted()
```

File: Tests/Source/Synchronization/StoreUpdateEventTests.swift [hunk 1]
```diff
@@ -29,8 +29,9 @@ class StoreUpdateEventTests: MessagingTest {
     override func setUp() {
         super.setUp()
         account = Account(userName: "John Doe", userIdentifier: UUID())
-        encryptionKeys = try! EncryptionKeys.createKeys(for: account)
-        publicKey = try! EncryptionKeys.publicKey(for: account)
+        //Notice: keys are nil when test with iOS 15 simulator. ref:https://wearezeta.atlassian.net/browse/SQCORE-1188
```
Comment (billypchan) [inline Tests/Source/Synchronization/StoreUpdateEventTests.swift:32]:
```suggestion
        // Notice: keys are nil when test with iOS 15 simulator. ref:https://wearezeta.atlassian.net/browse/SQCORE-1188
```

**Review by KaterinaWire — APPROVED** [review]

**Comment by billypchan** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @billypchan! Merging PR for command **release patch** for PR #**1619**. Will create a **patch** release after merged.
Next version is 405.0.1.

**Pull request merged**